staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -350,5 +352,9 @@ func ObserveWaitingDuration(ctx context.Context, priorityLevel, flowSchema, exec
 
 // ObserveExecutionDuration observes the execution duration for flow control
 func ObserveExecutionDuration(ctx context.Context, priorityLevel, flowSchema string, executionTime time.Duration) {
-	apiserverRequestExecutionSeconds.WithContext(ctx).WithLabelValues(priorityLevel, flowSchema).Observe(executionTime.Seconds())
+	watch := ""
+	if requestInfo, ok := apirequest.RequestInfoFrom(ctx); ok {
+		watch = strconv.FormatBool(requestInfo.Verb == "watch")
+	}
+	apiserverRequestExecutionSeconds.WithContext(ctx).WithLabelValues(priorityLevel, flowSchema, watch).Observe(executionTime.Seconds())
```

Comment:
we could include the verb in the metric, but that would increase the cardinality. I think for this metric we probably don't care beyond whether it's a watch.

Reply:
There are other long-running requests, but they currently do not contribute to this metric --- right?  Perhaps we should future-proof by calling this label "long-running"?

Reply:
You mean as a label in the metric? Yeah - that sgtm.

@tkashem - also - might be useful to split the metrics PR from the logs PR (the metrics will have release note, as suggest above, etc.).

Reply:
if we want to future-proof it (assuming that at some point apf may account for long-running requests), then i would suggest renaming the label to 'type' with values `{'regular'|'watch'|'non-watch-long-running'}` (for lack of better terms on my part) so we can differentiate between watch and non-watch long running requests.

Reply:
I like it - let's change the label name to "type" and for now use:
- "regular" [I'm not a fan of this name - but I don't have better alternative]
- "watch"

If at some point we start supporting other calls, we can introduce new values.

Reply:
I split the metrics into a separate PR - https://github.com/kubernetes/kubernetes/pull/105517 with release notes